gk/main.c

```diff
@@ -566,6 +566,21 @@ gk_flow_tbl_bucket_scan(uint32_t *bidx,
 				instance->ip_flow_hash_table, fe);
 		}
 
+		if (fe->state == GK_REQUEST) {
+			uint8_t priority = priority_from_delta_time(now,
+				fe->u.request.last_packet_seen_at);
+			/*
+			 * Remove request entries that are not doubling
+			 * its priority when a Gatekeeper server is overloaded.
+			 * We use +2 instead of +1 in the test below to account
+			 * for random delays in the network.
+			 */
+			if (priority > fe->u.request.last_priority + 2) {
```

Comment:
Shouldn't it be ``priority < fe->u.request.last_priority + 2`` (i.e., the request doesn't wait for long enough time)?

Also, shoudn't it be ``Remove request entries that are not doubling its waiting time`` instead of ``its priority`` in the comment?

Reply:
> Shouldn't it be ``priority < fe->u.request.last_priority + 2`` (i.e., the request doesn't wait for long enough time)?

The test is correct because it is testing for expiration. If a request entry has not seen another packet for more than the time needed to multiply its priority 4 times, we assume the client is no longer interested in establishing that flow.

> Also, shoudn't it be ``Remove request entries that are not doubling its waiting time`` instead of ``its priority`` in the comment?

I agree.

Reply:
Makes sense! Thanks